core-api/src/main/java/com/redhat/lightblue/Response.java

```diff
@@ -62,6 +76,31 @@ public Response(BaseResponse r) {
     }
 
     /**
+     * Result set size threshold is expressed in bytes. This is just an approximation, see @{link {@link JsonUtils#size(JsonNode)} for details.
+     *
+     * @param maxResultSetSizeB error when this threshold is breached
+     * @param warnResultSetSizeB log a warning when this threshold is breached
+     * @param forRequest request which resulted in this resposne, for logging purposes
+     */
+    public void ensureResponseSizeNotTooLarge(int maxResultSetSizeB, int warnResultSetSizeB, Request forRequest) {
+        this.forRequest = forRequest;
+        this.maxResultSetSizeB = maxResultSetSizeB;
+        this.warnResultSetSizeB = warnResultSetSizeB;
+    }
+
+    public boolean isErrorOnResposneSizeTooLarge() {
+        return maxResultSetSizeB > 0;
+    }
+
+    public boolean isWarnOnResponseSizeLarge() {
```

Comment:
Why one "ResponseSizeLarge" and the other "ResponseSizeTooLarge"?

Also, still typo in "Resposne"